web/blog/tenzir-v4.12/index.md

```diff
@@ -0,0 +1,176 @@
+blog post](reduce-cost-and-noise-with-deduplication).
+
+Building on this, the
+[`every`](/language/operator-modifiers#scheduled-executions) operator
+(prominently featured in the [previous
+release](tenzir-v4.11#execute-sources-on-a-schedule)) can now also accompany
+transformations and sinks. To illustrate, let's answer this question: "With how
+many new destinations did each device communicate in the last minute?"
+
+Using `every 1min summarize num=count(.) by src`, we get:
+
+```json
+{"src": "192.0.2.1", "num": 2}
+// after 1min
+{"src": "192.0.2.1", "num": 1}
+```
+
+In summary, these transformations provide powerful in-band capabilities leading
+to substantial data reduction, simplifying the analysis, and making data shaping
+more efficient.
+
+## Publish and subscribe
+
+Exciting are also the new [`publish`](/next/operators/publish) and
+[`subscribe`](/next/operators/publish) operators, which open up endless
+possibilities for creating arbitrary dataflow topologies. For instance,
+you can set a publishing point within your data stream, as simple as `from
+tcp://0.0.0.0:8000 | publish input`. This defines a channel `input` that you can
+now subscribe to with `subscribe`.
+
+Let's consider a case where we aim to route all alerts into Splunk, and concurrently import all other non-alert events into Tenzir's storage for further analysis and monitoring:
+
+``` title="1st subscriber"
+subscribe input
+| where alert == true
+| to splunk
+```
+
+``` title="2nd subscriber"
+subscribe input
+| where alert == false
+| import
+```
+
+Here, the first subscriber takes the events from the `input` channel where the
+alert field is `true` and routes them to Splunk. In parallel, the second
+subscriber takes the remaining events with the alert field marked as `false` and
+imports them into Tenzir’s storage. Our new feature enables this precise,
+dynamic routing, making data management more efficient and streamlined.
+
+## Contexts as Code
+
+[Tenzir v4.10](tenzir-v4.10) introduced introduced the ability to statically
+define pipelines in Tenzir's configuration file: **Pipelines as Code (PaC)**.
+This release expands upon that capability by also allowing static configuration
+of contexts.
+
+```yaml title="tenzir.yaml"
+tenzir:
+  contexts:
+    # A unique name for the context that's used in the context, enrich, and
+    # lookup operators to refer to the context.
+    indicators:
+      # The type of the context (e.g., `lookup-table`, `geoip`, ...).
+      type: bloom-filter
+      # Arguments for creating the context, as described by the documentation of
+      # the chosen context type.
+      arguments:
+        capacity: 1B
+        fp-probability: 0.001
+```
+
+## Download Button
+
+Ever wanted to just save the output from Explorer to your computer? With the new
+**Download** button, you can do precisely that:
+
+![Download Button](download-button.png)
+
+Just select an available form and you're good to go!
+
+## Other Changes
```

Comment:
Here or above we should prominently mention the potentially breaking change to `context update` and others.